### PR TITLE
 Allow a borrowed address to be passed as in_guaranteed arg in partial_apply [on_stack] only

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci smoke test

File: test/SIL/ownership-verifier/load_borrow_invalidation_partial_apply.sil [hunk 1]
```diff
@@ -25,18 +25,6 @@ bb0(%0 :  $*WrapperStruct):
   return %res : $()
 }
 
-sil [ossa] @test2 : $@convention(thin) (@in_guaranteed WrapperStruct) -> () {
-bb0(%0 :  $*WrapperStruct):
-  %1 = load_borrow %0 : $*WrapperStruct
```
Comment (gottesmm) [inline test/SIL/ownership-verifier/load_borrow_invalidation_partial_apply.sil:30]: Can you add a positive test that checks that we fail.

**Comment by meg-gupta** [comment]
@swift-ci smoke test

**Comment by meg-gupta** [comment]
@swift-ci smoke test

**Pull request merged**